rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/TEContentProvider.xtend

```diff
@@ -13,26 +13,30 @@
 package org.testeditor.rcp4.views.projectexplorer
 
 import org.eclipse.core.resources.IProject
-import org.eclipse.core.resources.ResourcesPlugin
+import org.eclipse.core.resources.IResource
 import org.eclipse.jdt.core.IClasspathEntry
 import org.eclipse.jdt.core.JavaCore
 import org.eclipse.jface.viewers.ITreeContentProvider
 import org.eclipse.jface.viewers.Viewer
+import org.eclipse.xtend.lib.annotations.Accessors
 
 /**
  * Content Provider to extend the cnf navigator with the TE specific elements. The provider replaces folders of the classpath entry with one entry.
  */
 class TEContentProvider implements ITreeContentProvider {
 
+	@Accessors WorkspaceRootHelper workspaceRootHelper = new WorkspaceRootHelper
```

Comment:
Isn't the default e4 injection working here? Otherwise we can implement it like the `rcp4.views.tcltestrun`, see `org.testeditor.rcp4.views.tcltestrun.ExecutableExtensionFactory`